python/ray/_raylet.pyx

```diff
@@ -2569,11 +2569,10 @@ def maybe_initialize_job_config():
         # to sys.path.
         this_node_id = ray._private.worker.global_worker.current_node_id
         driver_node_id = ray.NodeID(core_worker.get_job_config().driver_node_id)
-        if this_node_id == driver_node_id:
-            py_driver_sys_path = core_worker.get_job_config().py_driver_sys_path
-            if py_driver_sys_path:
-                for p in py_driver_sys_path:
-                    sys.path.insert(0, p)
+        py_driver_sys_path = core_worker.get_job_config().py_driver_sys_path
+        if py_driver_sys_path:
+            for p in py_driver_sys_path:
+                sys.path.insert(0, p)
```

Comment:
we need to change this to `sys.path.append(p)` to make sure the added sys path is placed in the last searched order, to avoid it overwriting working_dir, if any.

Reply:
Append actually won't work unfortunately

Reply:
Ray won't be able to start if use append
<img width="1002" alt="image" src="https://github.com/ray-project/ray/assets/7553988/89de983c-fe6a-44d4-b101-f6583e40dee1">

Reply:
I am confused. does this mean we have a different Ray in multiple places?

Reply:
should be the same ray. But apparently the order matters